### ISSUE TITLE
CRITICAL Failed to find driver molecule_hetznercloud. Please ensure that the driver is correctly installed.

### ISSUE THREAD

**Issue by jakoberpf** [ansible-community/molecule-hetznercloud#61]
I have problems using the module with my molecule setup. Seems more related to my `pip` but I am clueless what is wrong. Maybe someone can tell me whats wrong.

I installed the plugin with `pip install --upgrade molecule-hetznercloud` and also `pip3 install --upgrade molecule-hetznercloud` to be sure, but when I run `molecule test` I get the following errors.

```py
Failed to load driver entry point Traceback (most recent call last):
  File "/Users/jakoberpf/Library/Python/3.11/lib/python/site-packages/molecule/api.py", line 53, in drivers
    pm.load_setuptools_entrypoints("molecule.driver")
  File "/opt/homebrew/lib/python3.11/site-packages/pluggy/_manager.py", line 288, in load_setuptools_entrypoints
    self.register(plugin, name=ep.name)
  File "/opt/homebrew/lib/python3.11/site-packages/pluggy/_manager.py", line 91, in register
    raise ValueError(
ValueError: Plugin already registered: delegated=<class 'molecule.driver.delegated.Delegated'>
{'default': <class 'molecule.driver.delegated.Delegated'>, 'azure': <class 'molecule_plugins.azure.driver.Azure'>, 'containers': <class 'molecule_plugins.containers.driver.Container'>, 'docker': <class 'molecule_plugins.docker.driver.Docker'>, 'ec2': <class 'molecule_plugins.ec2.driver.EC2'>, 'gce': <class 'molecule_plugins.gce.driver.GCE'>, 'podman': <class 'molecule_plugins.podman.driver.Podman'>, 'vagrant': <class 'molecule_plugins.vagrant.driver.Vagrant'>}

ERROR    Failed to load driver entry point Traceback (most recent call last):
  File "/Users/jakoberpf/Library/Python/3.11/lib/python/site-packages/molecule/api.py", line 53, in drivers
    pm.load_setuptools_entrypoints("molecule.driver")
  File "/opt/homebrew/lib/python3.11/site-packages/pluggy/_manager.py", line 288, in load_setuptools_entrypoints
    self.register(plugin, name=ep.name)
  File "/opt/homebrew/lib/python3.11/site-packages/pluggy/_manager.py", line 91, in register
    raise ValueError(
ValueError: Plugin already registered: delegated=<class 'molecule.driver.delegated.Delegated'>
{'default': <class 'molecule.driver.delegated.Delegated'>, 'azure': <class 'molecule_plugins.azure.driver.Azure'>, 'containers': <class 'molecule_plugins.containers.driver.Container'>, 'docker': <class 'molecule_plugins.docker.driver.Docker'>, 'ec2': <class 'molecule_plugins.ec2.driver.EC2'>, 'gce': <class 'molecule_plugins.gce.driver.GCE'>, 'podman': <class 'molecule_plugins.podman.driver.Podman'>, 'vagrant': <class 'molecule_plugins.vagrant.driver.Vagrant'>}

CRITICAL Failed to find driver molecule_hetznercloud. Please ensure that the driver is correctly installed.
```

From the `molecule --version` command I am guessing that the driver is not correctly installed but am clueless what the issue is. I would be very thankfull about any hints.

```ruby
molecule 6.0.2 using python 3.11 
    ansible:2.15.8
    azure:23.5.0 from molecule_plugins
    containers:23.5.0 from molecule_plugins requiring collections: ansible.posix>=1.3.0 community.docker>=1.9.1 containers.podman>=1.8.1
    default:6.0.2 from molecule
    docker:23.5.0 from molecule_plugins requiring collections: community.docker>=3.0.2 ansible.posix>=1.4.0
    ec2:23.5.0 from molecule_plugins
    gce:23.5.0 from molecule_plugins requiring collections: google.cloud>=1.0.2 community.crypto>=1.8.0
    podman:23.5.0 from molecule_plugins requiring collections: containers.podman>=1.7.0 ansible.posix>=1.3.0
    vagrant:23.5.0 from molecule_plugins
```

**Comment by jooola** [comment]
Hi @jakoberpf 

Did you by any chance unintentionally upgrade from the version 1.x to version 2.x ? If did you follow the upgrading guide (https://github.com/ansible-community/molecule-hetznercloud#upgrading-to-v2) ?

What does `pip freeze` return?

**Comment by jooola** [comment]
Could you also provide me with the molecule configuration files, so I can try to reproduce your issue ?

**Comment by jakoberpf** [comment]
@jooola Happy to.

```yaml
---
driver:
  name: molecule_hetznercloud
platforms:
  - name: instance-1
    image: debian-12
    server_type: cx11
    location: hel1
    volumes:
      - name: volume-1
        size: 10
  - name: instance-2
    image: debian-12
    server_type: cx11
    location: hel1
    volumes:
      - name: volume-1
        size: 10
  - name: instance-3
    image: debian-12
    server_type: cx11
    location: hel1
    volumes:
      - name: volume-1
        size: 10
```

**Comment by jooola** [comment]
Hmmm, I cannot reproduce your issue.

Could you try installing only the minimal dependencies in a fresh venv, and check if the molecule_hetznercloud plugin is detected?

```
python3 -m venv venv
source venv/bin/activate
pip install molecule==6.0.3 molecule-hetznercloud==2.0.0
molecule --version
```

Output:
```
molecule 6.0.3 using python 3.11 
    ansible:2.16.2
    default:6.0.3 from molecule
    molecule_hetznercloud:2.0.0 from molecule_hetznercloud
```

**Comment by jakoberpf** [comment]
Mhh just executes you command and am getting this output. I have a couple more dependencies installed as it looks like...

```bash
python3 -m venv venv
source venv/bin/activate
pip install molecule==6.0.3 molecule-hetznercloud==2.0.0
molecule --version
......
Installing collected packages: resolvelib, netaddr, urllib3, subprocess-tee, six, rpds-py, PyYAML, pygments, pycparser, pluggy, packaging, mdurl, MarkupSafe, idna, click, charset-normalizer, certifi, bracex, attrs, wcmatch, requests, referencing, python-dateutil, markdown-it-py, Jinja2, click-help-colors, cffi, rich, jsonschema-specifications, cryptography, jsonschema, enrich, ansible-core, ansible-compat, molecule, molecule-hetznercloud
Successfully installed Jinja2-3.1.2 MarkupSafe-2.1.3 PyYAML-6.0.1 ansible-compat-4.1.10 ansible-core-2.16.2 attrs-23.2.0 bracex-2.4 certifi-2023.11.17 cffi-1.16.0 charset-normalizer-3.3.2 click-8.1.7 click-help-colors-0.9.4 cryptography-41.0.7 enrich-1.2.7 idna-3.6 jsonschema-4.20.0 jsonschema-specifications-2023.12.1 markdown-it-py-3.0.0 mdurl-0.1.2 molecule-6.0.3 molecule-hetznercloud-2.0.0 netaddr-0.10.1 packaging-23.2 pluggy-1.3.0 pycparser-2.21 pygments-2.17.2 python-dateutil-2.8.2 referencing-0.32.1 requests-2.31.0 resolvelib-1.0.1 rich-13.7.0 rpds-py-0.16.2 six-1.16.0 subprocess-tee-0.4.1 urllib3-2.1.0 wcmatch-8.5

[notice] A new release of pip is available: 23.3.1 -> 23.3.2
[notice] To update, run: pip install --upgrade pip
Failed to load driver entry point Traceback (most recent call last):
  File "/Users/jakoberpf/Library/Python/3.11/lib/python/site-packages/molecule/api.py", line 53, in drivers
    pm.load_setuptools_entrypoints("molecule.driver")
  File "/opt/homebrew/lib/python3.11/site-packages/pluggy/_manager.py", line 288, in load_setuptools_entrypoints
    self.register(plugin, name=ep.name)
  File "/opt/homebrew/lib/python3.11/site-packages/pluggy/_manager.py", line 91, in register
    raise ValueError(
ValueError: Plugin already registered: delegated=<class 'molecule.driver.delegated.Delegated'>
{'default': <class 'molecule.driver.delegated.Delegated'>, 'azure': <class 'molecule_plugins.azure.driver.Azure'>, 'containers': <class 'molecule_plugins.containers.driver.Container'>, 'docker': <class 'molecule_plugins.docker.driver.Docker'>, 'ec2': <class 'molecule_plugins.ec2.driver.EC2'>, 'gce': <class 'molecule_plugins.gce.driver.GCE'>, 'podman': <class 'molecule_plugins.podman.driver.Podman'>, 'vagrant': <class 'molecule_plugins.vagrant.driver.Vagrant'>}

molecule 6.0.2 using python 3.11 
    ansible:2.16.2
    azure:23.5.0 from molecule_plugins
    containers:23.5.0 from molecule_plugins requiring collections: ansible.posix>=1.3.0 community.docker>=1.9.1 containers.podman>=1.8.1
    default:6.0.2 from molecule
    docker:23.5.0 from molecule_plugins requiring collections: community.docker>=3.0.2 ansible.posix>=1.4.0
    ec2:23.5.0 from molecule_plugins
    gce:23.5.0 from molecule_plugins requiring collections: google.cloud>=1.0.2 community.crypto>=1.8.0
    podman:23.5.0 from molecule_plugins requiring collections: containers.podman>=1.7.0 ansible.posix>=1.3.0
    vagrant:23.5.0 from molecule_plugins
```

**Comment by jooola** [comment]
This issue is probably related to `pluggy`, the plugin manager used by molecule.

Could you open a ticket against the main molecule repo? I do not think this is related to our plugin.

I would also check that the venv is clean, without external dependency injected.

You could also hunt down why you have a 2 delegated plugin being registered. According to https://packaging.python.org/en/latest/guides/creating-and-discovering-plugins/#using-package-metadata, you can list all the packages that have a `molecule.driver` entry point using:

```
venv/bin/python
```
```py
from importlib.metadata import entry_points

discovered_plugins = entry_points(group="molecule.driver")
print(discovered_plugins)
```
